### DISCUSSION TITLE
Subscript

### DISCUSSION THREAD

**Discussion by wartburggraaf** [nl-design-system/backlog#334] [Component suggestions]
<img width="480" alt="visuele weergave van een Subscript component" src="https://github.com/user-attachments/assets/af5f4f71-a4cc-42c6-a871-a96482b2b6b8">

We willen graag meer leren over de Subscript component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1725519600501559)

## Naam

Subscript

## Doel

Toont tekst met een kleiner lettertype met een verlaagde letterlijn.

## Gerelateerde componenten

[Superscript](https://github.com/orgs/nl-design-system/discussions/336)

## Onderzoek
[Subscript and superscript - Wikipedia](https://en.wikipedia.org/wiki/Subscript_and_superscript)
Wikipedia pagina waarbij ook vele voorbeelden van gebruik worden beschreven.

[`<sub>`: The Subscript element - MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/sub)
Documentatie van Mozilla Developers Network over het `<sub>` HTML element.

[Problems solved by OpenType | Roel Nieskens | CSS Day 2024 - Youtube](https://www.youtube.com/watch?v=TreBK-EyACQ&t=1520s)
Riel Nieskens deelt een presentatie bordevol tips en toont onder andere hoe je super- en subscript in CSS kan gebruiken.

[Brief Note on Super- and Subscript Text - Adrian Roselli](https://adrianroselli.com/2022/09/brief-note-on-super-and-subscript-text.html)
Adrian Roselli heeft onderzoek gedaan naar support voor Super- en Subscript door assistieve technologie.

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-subscript--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/subscript)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-09-27 om 13 26 15](https://github.com/user-attachments/assets/23569630-5161-4bbf-a449-c515cc40279b)

Subscript inside multiple lines of text
![Scherm­afbeelding 2024-09-27 om 13 26 21](https://github.com/user-attachments/assets/92edfdd4-82fa-45e2-aadf-903155e48dc5)

Subscript with Open Type rendering, inside multiple lines of text
![Scherm­afbeelding 2024-09-27 om 13 26 28](https://github.com/user-attachments/assets/4e87928c-54e4-4de2-b05d-97a90ea7630d)


---

### Notities

Documentatie aanwezig (Storybook)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Subscript op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=subscript)

**Comment by rianrietveld** [comment]
Klein onderzoekje naar screenreader feedback op` <sub>`.
Zie ook vergelijkbare tekst bij de [sup-discussie](https://github.com/orgs/nl-design-system/discussions/336).

TL;DR: screenreaders geven geen info over sub of sub.

Alleen VoiceOver probeert er wat van het maken als deze de structuur herkent.
Dus `12 m<sup>2</sup>` wordt als twaalf vierkante meter uitgesproken. Maar `3<sup>2</sup>` wordt voorgelezen als 32.
NVDA wacht even voor het uitspreken van de sub en sup maar noemt alleen het nummer op.
JAWS negeert het volkomen en maakt het onderdeel van de tekst.
Dit valt dus in dezelfde categorie als `<b>` em `<i>`.
Alleen visuele informatie die niet wordt doorgegeven aan screen readers.
Ik zou deze component afraden voor content. 
Zie voor testcode:
https://codepen.io/rianrietveld/pen/xxvPEar

Discussie met @Robbert
unicode karakters als m². VoiceOver leest 12² voor als “12 digit 2”.  Zonder de sup-info.
We moeten de veelvoorkomende use cases uitsplitsen, en een advies moeten geven per onderdeel zoals:
- chemische formules: H2O
- wiskundige formules
[á la Julesprudentie](https://www.200ok.nl/tips/speciale-tekens-en-screenreaders/)
H2O lijkt me bijvoorbeeld harmless, voor formules moeten we misschien figures + figcaption + mathml adviseren

Hidde:
Er zijn wel mappings in HTML-AAM ([sub](https://www.w3.org/TR/html-aam-1.0/#el-sub) en [sup](https://www.w3.org/TR/html-aam-1.0/#el-sup)), dus dit zijn screenreader bugs. Adrian heeft een [issue gefiled bij JAWS](https://github.com/FreedomScientific/standards-support/issues/668), lijkt weinig te gebeuren.

Nadeel van geen component is denk ik dat als in de toekomst support beter wordt we geen updates aan een component kunnen doen? Maar niet echt een nadeel als het alternatief 'uitschrijven' is en dat ook gewoon goed genoeg werkt
